ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -247,6 +424,14 @@ KEXT_STATIC int HandleVnodeOperation(
     char procname[MAXCOMLEN + 1] = "";
     bool isDeleteAction = false;
     bool isDirectory = false;
+    bool isRename = true;
+
+    isDeleteAction = ActionBitIsSet(action, KAUTH_VNODE_DELETE);
+    if (isDeleteAction && VnodeIsEligibleForEventHandling(currentVnode))
```

Comment:
Can we do this check after the ShouldHandleVnodeOpEvent?

Then I don't think we'd need a VnodeIsEligibleForEventHandling check.

Reply:
We cannot move this because `DeleteOpIsForRename` clears the entry from the array, so if we didn’t call it the entries would start piling up.

Reply:
Is the concern `ShouldHandleVnodeOpEvent` may fail for another reason and we'd miss calling `DeleteOpIsForRename`

If having the check here is a must I'd suggest doing an explicit `VnodeIsEligibleForEventHandling` check for all nodes and removing it from `ShouldHandleVnodeOpEvent`.  I think that may be easier to understand.  

Some comments would be great too :-)

Reply:
> Is the concern `ShouldHandleVnodeOpEvent` may fail for another reason and we'd miss calling `DeleteOpIsForRename`

Precisely.

> If having the check here is a must I'd suggest doing an explicit `VnodeIsEligibleForEventHandling` check for all nodes and removing it from `ShouldHandleVnodeOpEvent`. I think that may be easier to understand.

That is absolutely a change I was planning to make, but as part of #900. As it's fairly standalone I can apply it here though.

> Some comments would be great too :-)

OK, will do.

Reply:
> That is absolutely a change I was planning to make, but as part of #900. As it's fairly standalone I can apply it here though.

I think it would be great to apply here if you don't mind.  It makes me it easier for me at least to read :-)

Reply:
I've now done that, as a separate commit in the series as it required an additional test to maintain 100% coverage and some updates to perf sampling which are completely unrelated to #526.